internal/generator/vector/output/cloudwatch/cloudwatch_test.go

```diff
@@ -0,0 +1,183 @@
+package cloudwatch
+
+import (
+	"testing"
+
+	"github.com/openshift/cluster-logging-operator/internal/generator"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/output/security"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ = Describe("Generate Vector config", func() {
+	inputPipeline := []string{"application-logs"}
+	prefix := "all-logs"
+	var f = func(clspec logging.ClusterLoggingSpec, secrets map[string]*corev1.Secret, clfspec logging.ClusterLogForwarderSpec, op generator.Options) []generator.Element {
+		return Conf(clfspec.Outputs[0], inputPipeline, secrets[clfspec.Outputs[0].Name], op)
+	}
+	DescribeTable("For Cloudwatch output", generator.TestGenerateConfWith(f),
+		Entry("with auth.access_key_id auth.secret_access_key auth.credentials_file", generator.ConfGenerateTest{
+			CLFSpec: logging.ClusterLogForwarderSpec{
+				Outputs: []logging.OutputSpec{
+					{
+						Type: logging.OutputTypeCloudwatch,
+						Name: "cw",
+						URL:  "https://cw.svc.all.cluster:9200",
+						OutputTypeSpec: logging.OutputTypeSpec{
+							Cloudwatch: &logging.Cloudwatch{
+								Region:      "us-east-1",
+								GroupBy:     "logType",
+								GroupPrefix: &prefix,
+							},
+						},
+
+						Secret: &logging.OutputSecretSpec{
+							Name: "cw",
+						},
+					},
+				},
+			},
+			Secrets: map[string]*corev1.Secret{
+				"cw": {
+					Data: map[string][]byte{
+						"aws_access_key_id":     []byte("xXyYzZ"),
+						"aws_secret_access_key": []byte("sSxXyYzZ"),
+					},
+				},
+			},
+			ExpectedConf: `
+			# Adding group_name and stream_name field
+[transforms.cw_add_grpandstream]
```

Comment:
@vimalk78 please advise how you are naming routing sections.  should it be something like `normalize_cw` or similar

Reply:
perhaps it can named `cw_routing`as it is routing messages to cw logstrems?